luigi/contrib/bigquery.py

```diff
@@ -118,14 +118,23 @@ def __init__(self, oauth_credentials=None, descriptor='', http_=None):
 
     def dataset_exists(self, dataset):
         """Returns whether the given dataset exists.
+        If regional location is specified for the dataset, that is also checked
+        to be compatible with the remote dataset, otherwise an exception is thrown.
 
            :param dataset:
            :type dataset: BQDataset
         """
 
         try:
-            self.client.datasets().get(projectId=dataset.project_id,
-                                       datasetId=dataset.dataset_id).execute()
+            response = self.client.datasets().get(projectId=dataset.project_id,
+                                                  datasetId=dataset.dataset_id).execute()
+            if dataset.location is not None:
+                fetched_location = response.get('location', '')
+                if not fetched_location:
+                    fetched_location = 'undefined'
```

Comment:
If we want `'undefined'` as the default value then we should specify that instead of `''`...